library_conventions/packages.adoc

```diff
@@ -0,0 +1,232 @@
+metal tab down to the PCB, or you may want to solder the tab to the PCB without
+a hole in it. For all these cases *only one Package is needed* -- the different
+mounting variants should be handled by different footprints inside the Package.
+====
+
+
+[#libraryconventions-packages-naming]
+==== Naming
+
+Following conventions apply to Package names:
+
+* *Language must be American English* (en_US), if applicable (many Packages
+  have language-neutral names anyway).
+* *If {ipc7351-naming-pdf}[IPC-7351] specifies a name for a given package, use
+  that name* (e.g. "SOT95P280X145-5" instead of "SOT23-5"). Alternative names
+  (like "SOT23-5") should be added to the comma-separated keywords list.
+* *Size information must be given in millimeters*, not inches or mils. For
+  packages which are well known by their size in inches (e.g. "0805" which is
+  "2012" in metric), it's recommended to write the well known name in
+  parentheses. For example a resistor could be called "RESC2012X70 (0805)".
+* For packages which are available with different pin counts, *append the pin
+  count with a hyphen as separator and omit leading zeros* (e.g. "DIP-8" instead
+  of "DIP08").
+* For manufacturer-specific packages, start with the manufacturers name
+  (e.g. "Molex 53261-06").
+
+
+[#libraryconventions-packages-pads]
+==== Pads
+
+* *Always add all pads of Packages*, even those which are not connected in
+  some parts! For example the package "TO220" has a metal tab, so you should
+  define it as a pad, no matter if it's often not connected (and even not
+  connectable when mounted vertically).
+* *Use pad names according {ipc7351-pdf}[IPC-7351]* (if applicable). For
+  Packages which are not covered by IPC-7351:
+** If the function of a pad is absolutely clear, choose its abbreviated
+   functionality as name (e.g. "A" for anode and "C" for cathode).
+** Otherwise just use numbers starting with "1" at top left and increment
+   them counterclockwise.
+
+
+[#libraryconventions-packages-footprints]
+==== Footprint Variants
+
+Within a Package there can be multiple footprint variants. They are intended
+to support following use-cases:
+
+* *Mounting variants*: For example a THT resistor can be mount either
+  vertically or horizontally with various pad distance. Every mounting
+  variant should be available as footprint variants.
+* *Soldering techniques*: Many packages can be soldered either by reflow-,
+  wave- or hand-soldering, which usually require different land patterns. For
+  every suitable soldering technique there should be a corresponding footprint
+  variant.
+* *Density levels*: {ipc7351-pdf}[IPC-7351] specifies three different density
+  levels for footprints:
++
+--
+* Density Level A: Maximum (Most) Land Protrusion
+* Density Level B: Median (Nominal) Land Protrusion
+* Density Level C: Minimum (Least) Land Protrusion
+--
+If applicable, these three density levels should also be added as footprint
+variants.
```

Comment:
> Are you sure that you want to require both density levels and soldering techniques? That would mean that a part with hand soldering and reflow soldering variants plus all density levels would result in six combinations...

No, the idea is not to have 3 hand soldering footprints and 3 reflow soldering footprints. This section is meant to describe what's the purpose of the footprint variants, and soldering techniques and density levels are two possible use-cases.

But of course it's allowed to define for example 3 footprints for density level A/B/C, where density level A is also considered as hand-soldering variant. Or one more footprint with even larger pads than level A is added especially for hand soldering. Hand soldering is basically just a special density level with pad sizes suitable for hand soldering.

> But in the end, if we name the footprint variants "Density Level A" or "Density Level B" then people won't really know intuitively what this means. I think I'd favor naming like "Reflow", "Hand Soldering" or "Compact".

I agree that it's not intuitive for users not knowing IPC, but in the industry IPC is very well known and these users should immediately see the density levels because they know what level fits their needs. Maybe a compromise would be good, for example:

- "Density Level A (large)"
- "Density Level B (medium)"
- "Density Level C (small)"

Or if hand-soldering should have different pad sizes than Level A:

- "Density Level A (large)"
- "Density Level B (medium)"
- "Density Level C (small)"
- "Hand Soldering"

Reply:
Ok, you're right. I mis-read the introductory paragraph.

But maybe the wording could be a bit less normative? Right now it says:

> For every suitable soldering technique there should be a corresponding footprint variant.

We could change "should" to "could", since we don't want to *require* different footprint variants for hand soldering, reflow soldering, hot-air soldering, wave soldering, laser reflow soldering, vapor phase soldering and whatever other crazy soldering methods there are :slightly_smiling_face:

Reply:
> We could change "should" to "could"

Done.